pages/map.tsx

```diff
@@ -39,27 +46,77 @@ function Map() {
     })   
   },[map])
 
-  const addZone = useCallback((e:google.maps.MapMouseEvent) =>{
+  const addZone = (e:google.maps.MapMouseEvent) =>{
     const geocoder = new google.maps.Geocoder();
-    const location= e.latLng as GoogleLatLng;
-
+    const location = {
+      lat: e.latLng!.lat(),
+      lng: e.latLng!.lng(),
+    };
+    
     geocoder.geocode({ location }, (results, status) => {
         if (status === "OK" && results) {
-          // to do open modal
+          new google.maps.Marker({
+            position: new google.maps.LatLng(location),
+            map: map as any,
+          });
+          
+          setPartialZone({
+            address: {
+              name: results[0].formatted_address,
+              lat: location.lat,
+              lng: location.lng
+            },
+            volunteerName: username
+          });
         }  
     });
-  },[])
+    setAddModalVisible(true);
+  }
+
+  const findZoneByCoordinates = (coordinates: any): InterestZone | undefined => {
+    return interestZones.find((zone: InterestZone) => {
+      return zone.address.lat === coordinates.lat && zone.address.lng === coordinates.lng;
+    });
+  }
+
+  const closeEditModal = useCallback(() => {
+    setEditModalVisible(false);
+  }, [editModalVisible]);
+
+  const openEditModal = useCallback(() => {
+    setEditModalVisible(true);
+  }, [editModalVisible]);
+
+  const displayZoneMarker = (e: google.maps.MapMouseEvent): void => {
+      const lat = e.latLng!.lat();
+      const lng = e.latLng!.lng();
+
+      const interestZone = findZoneByCoordinates({lat: lat, lng: lng});
```

Comment:
In theory, this should not be working since the `findZoneByCoordinates` method returns a promise and not a zone. Please move the logic directly here and update the state within the then block.

P.S. try to avoid defining methods within a component, that are not fit for react hooks (useEffect, useCallback)